components/location/listItem.jsx

```diff
@@ -0,0 +1,75 @@
+/* eslint-disable react/prop-types */
+import React, { useEffect, useState } from "react";
+
+const LocationListItem = ({
+  closePopups,
+  index,
+  location,
+  mapObject,
+  toggleLocationList,
+}) => {
+  const [onClick, setOnClick] = useState(null);
+
+  useEffect(() => {
+    if (location.marker && !onClick) {
+      // Reason why this arrow function looks so weird: https://stackoverflow.com/a/55621325
```

Comment:
Hmm yeah I need to come back to this

Reply:
Yeah that comment is somewhat incomplete (sorry about that), but that's the syntax required if your setting a state variable to a function!